### ISSUE TITLE
Uncaught Error: QMX_Collector_Image_Sizes::filter__wp_get_attachment_image_src(): Argument #2 ($attachment_id) must be of type int, string given

### ISSUE THREAD

**Issue by brambil** [crstauf/query-monitor-extend#77]
Removing WooCommerce Order using bulk action:

`Fatal error: Uncaught Error: QMX_Collector_Image_Sizes::filter__wp_get_attachment_image_src(): Argument #2 ($attachment_id) must be of type int, string given, called in /home/XXXXXXX/public_html/wp-includes/class-wp-hook.php on line 326
in /home/XXXXXXX/public_html/wp-content/plugins/query-monitor-extend/image-sizes/qmx-image-sizes-collector.php on line 148

Call stack:

QMX_Collector_Image_Sizes::filter__wp_get_attachment_image_src()
wp-includes/class-wp-hook.php:326
WP_Hook::apply_filters()
wp-includes/plugin.php:205
apply_filters()
wp-includes/media.php:1008
wp_get_attachment_image_src()
wp-includes/media.php:1179
wp_get_attachment_image_url()
wp-content/plugins/woocommerce/includes/class-wc-structured-data.php:488
WC_Structured_Data::generate_order_data()
wp-includes/class-wp-hook.php:326
WP_Hook::apply_filters()
wp-includes/class-wp-hook.php:348
WP_Hook::do_action()
wp-includes/plugin.php:517
do_action()
wp-content/plugins/woocommerce/templates/emails/admin-cancelled-order.php:37
include()
wp-content/plugins/woocommerce/includes/wc-core-functions.php:345
wc_get_template()
wp-content/plugins/woocommerce/includes/wc-core-functions.php:364
wc_get_template_html()
wp-content/plugins/woocommerce/includes/emails/class-wc-email-cancelled-order.php:105
WC_Email_Cancelled_Order::get_content_html()
wp-content/plugins/woocommerce/includes/emails/class-wc-email.php:588
WC_Email::get_content()
wp-content/plugins/woocommerce/includes/emails/class-wc-email-cancelled-order.php:93
WC_Email_Cancelled_Order::trigger()
wp-includes/class-wp-hook.php:324
WP_Hook::apply_filters()
wp-includes/class-wp-hook.php:348
WP_Hook::do_action()
wp-includes/plugin.php:565
do_action_ref_array()
wp-content/plugins/woocommerce/includes/class-wc-emails.php:172
WC_Emails::send_transactional_email()
wp-includes/class-wp-hook.php:324
WP_Hook::apply_filters()
wp-includes/class-wp-hook.php:348
WP_Hook::do_action()
wp-includes/plugin.php:517
do_action()
wp-content/plugins/woocommerce/includes/class-wc-order.php:420
WC_Order::status_transition()
wp-content/plugins/woocommerce/includes/class-wc-order.php:253
WC_Order::save()
wp-content/plugins/woocommerce/includes/class-wc-order.php:365
WC_Order::update_status()
wp-content/plugins/woocommerce/includes/admin/list-tables/class-wc-admin-list-table-orders.php:477
WC_Admin_List_Table_Orders::handle_bulk_actions()
wp-includes/class-wp-hook.php:324
WP_Hook::apply_filters()
wp-includes/plugin.php:205
apply_filters()
wp-admin/edit.php:222`

[fatal-errors-2024-02-27.log](https://github.com/crstauf/query-monitor-extend/files/14420902/fatal-errors-2024-02-27.log)
[SystemStatusReport_www.XXXXX.it_2024-02-27T14-16-00.txt](https://github.com/crstauf/query-monitor-extend/files/14420903/SystemStatusReport_www.XXXXX.it_2024-02-27T14-16-00.txt)

**Comment by crstauf** [comment]
Unfortunately this is not an issue with QMX. Whatever code is printing an image is sending the attachment's ID as a string, rather than properly as an integer.

You can see in WooCommerce's documentation that they are returning the incorrect type: https://woocommerce.github.io/code-reference/classes/WC-Product.html#method_get_image_id.

I have reported to the WooCommerce team, but in my past experience they don't care much about these things:
https://github.com/woocommerce/woocommerce/issues/45168.

In the meantime, you can remove the `int` from the second parameter of `QMX_Collector_Image_Sizes::filter__wp_get_attachment_image_src()` to resolve the issue.

```php
public function filter__wp_get_attachment_image_src( $image, $attachment_id, $size ) {
	if ( did_action( 'qm/cease' ) ) {
		return $image;
	}
```

**Comment by chrillep** [comment]
acf did this for iconpicker

**Comment by crstauf** [comment]
Next published version will remove the type constraint.

(Resolved by a41cb14a1899bea5582e3d50efb61f1095913635.)
